Assets/_BForBoss/_Weapons/Scripts/Weapons/Weapon.cs

```diff
@@ -108,17 +121,18 @@ public Vector3 GetShootDirection(float timeSinceFiring)
              return new Vector3(xPosition, yPosition, 1);
          }
 
-         public void FireIfPossible()
+         public bool FireIfPossible()
```

Comment:
maybe a better method name would be `TryFire()` since its returning a bool. What do you think?

Reply:
TryFire makes sense to me, I'll rename it